feat(upload): 新增上传组件

HXCStudio123 opened the pull request:
### 新特性
- Upload
  - 新增上传组件
- ImgPreview
  - 新增图片预览组件

HXCStudio123 committed:
feat(upload): 新增上传组件

upload: 上传功能基本实现

fix: 上传功能补全

feat: 添加反转属性和自定义样式类

feat: 添加最大尺寸限制属性以及回调函数

HXCStudio123 committed:
feat: img-preview 图片预览组件

feat: 单击双击、长按、图片放大查看功能完成

feat: 图片预览添加 长按事件钩子 ，暴露图片预览打开关闭事件，样式变量抽取

feat: 模拟双指缩放

test: 真机调试

test: 持续调试

fix: 修复双指缩放结尾重置

docs: 图片预览文档补全

feat: upload 图片预览功能补全

yawuling commented on examples/docs/docs/upload.md:
```diff
@@ -0,0 +1,347 @@
+
+### 自定义预览
+
+设置 `before-preview` 函数，在用户点击图片进行预览时，会执行 `before-remopreviewve` 函数，并传入 `file` 和 `resolve` 参数，可以对 `file` 进行处理，并通过 `resolve` 函数告知组件是否确定通过，`resolve` 接受1个 boolean 值，`resolve(true)` 表示选项通过，`resolve(false)` 表示选项不通过，不通过时不会执行预览图片操作。
```
before-remopreviewve 写错了

HXCStudio123 committed:
icon: 新增icon

nothing: 无

fix: 修复cell bug，添加上传动画

fix: 修复引入路径问题

test: 真机测试 双指缩放优化

test: 真机测试

test: 持续测试

fix: 优化拖拽卡顿

fix: 真机测试

fix: 优化动画

fix: 上传 图片预览组件修复问题

style: 添加动画，属性名校准

fix: 动画复原

docs: 移动顺序

fix: pr问题修改

fix: 上传优化

fix: 优化动效

test: 测试

fix: 移除双指

fix: cover 问题修改

yawuling commented on packages/img-preview/src/main.vue:
```diff
@@ -0,0 +1,435 @@
+      if (this.swipeInitialIndex !== 0) {
+        // 初始化时将轮播滑动时间置为0，禁用跳转index动画
+        this.duration = 0
+        this.initSwipeTimer = setTimeout(() => {
```
改用nextTick，这里定时器被清了，怎么执行

yawuling commented on examples/docs/docs/imgPreview.md:
```diff
@@ -0,0 +1,175 @@
+  data () {
+    return {
+      imgList: [
+        'https://img12.360buyimg.com//n0/jfs/t1/29118/6/4823/55969/5c35c16bE7c262192/c9fdecec4b419355.jpg',
```
图片地址不对， n0 前面有2个 /
别用有赞的cdn

yawuling commented on examples/docs/docs/upload.md:
```diff
@@ -0,0 +1,358 @@
+| loading-color | [加载中图标颜色](/docs#/components/loading) | string | — | '#ffffff' |
+| loading-size | [加载中图标尺寸](/docs#/components/loading) | string | — | '24px' |
+| show-img-preview | 是否开启图片预览，上传组件中使用了[图片预览组件 JS 调用](/docs#/components/imgPreview)的方式 | boolean | — | false |
+| maxZoom | 开启图片预览，手势缩放时，最大缩放比例，详细使用说明查看[图片预览组件 JS 调用](/docs#/components/imgPreview) | Number | -  | 3 |
```
用连字符

yawuling commented on packages/img-preview/src/main.vue:
```diff
@@ -0,0 +1,439 @@
+        this.setMoving()
+      }
+      // if (this.moving || this.zooming) {
+      //   event.preventDefault()
```
哪来的event
